airflow/models/trigger.py

```diff
@@ -121,17 +121,24 @@ def clean_unused(cls, session: Session = NEW_SESSION) -> None:
         # Update all task instances with trigger IDs that are not DEFERRED to remove them
         for attempt in run_with_db_retries():
             with attempt:
-                session.query(TaskInstance).filter(
-                    TaskInstance.state != TaskInstanceState.DEFERRED, TaskInstance.trigger_id.isnot(None)
-                ).update({TaskInstance.trigger_id: None})
+                session.execute(
+                    update(TaskInstance)
+                    .where(
+                        TaskInstance.state != TaskInstanceState.DEFERRED, TaskInstance.trigger_id.isnot(None)
```

Comment:
```suggestion
                        TaskInstance.state != TaskInstanceState.DEFERRED, TaskInstance.trigger_id.is_not(None)
```

`isnot` is not formally deprecated but `is_not` is favored

Reply:
fixed